hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -0,0 +1,604 @@
+    OFSPath ofsPath = new OFSPath(pathStr);
+    String keyStr = ofsPath.getKeyName();
+    try {
+      OzoneBucket bucket = getBucket(ofsPath, true);
+      bucket.createDirectory(keyStr);
+    } catch (OMException e) {
+      if (e.getResult() == OMException.ResultCodes.FILE_ALREADY_EXISTS) {
+        throw new FileAlreadyExistsException(e.getMessage());
+      }
+      throw e;
+    }
+    return true;
+  }
+
+  /**
+   * Helper method to delete an object specified by key name in bucket.
+   *
+   * @param path path to a key to be deleted
+   * @return true if the key is deleted, false otherwise
+   */
+  @Override
+  public boolean deleteObject(String path) {
+    LOG.trace("issuing delete for path to key: {}", path);
+    OFSPath ofsPath = new OFSPath(path);
+    String keyName = ofsPath.getKeyName();
+    try {
+      OzoneBucket bucket = getBucket(ofsPath, false);
+      incrementCounter(Statistic.OBJECTS_DELETED);
+      bucket.deleteKey(keyName);
+      return true;
+    } catch (IOException ioe) {
+      LOG.error("delete key failed " + ioe.getMessage());
+      return false;
+    }
+  }
+
+  public FileStatusAdapter getFileStatus(String path, URI uri,
+      Path qualifiedPath, String userName)
+      throws IOException {
+    OFSPath ofsPath = new OFSPath(path);
+    String key = ofsPath.getKeyName();
+    try {
+      OzoneBucket bucket = getBucket(ofsPath, false);
+      incrementCounter(Statistic.OBJECTS_QUERY);
+      OzoneFileStatus status = bucket.getFileStatus(key);
+      // Note: qualifiedPath passed in is good from
+      //  BasicRootedOzoneFileSystem#getFileStatus. No need to prepend here.
+      makeQualified(status, uri, qualifiedPath, userName);
+      return toFileStatusAdapter(status);
+
+    } catch (OMException e) {
+      if (e.getResult() == OMException.ResultCodes.FILE_NOT_FOUND) {
+        throw new
+            FileNotFoundException(key + ": No such file or directory!");
+      }
+      throw e;
+    }
+  }
+
+  public void makeQualified(FileStatus status, URI uri, Path path,
+      String username) {
+    if (status instanceof OzoneFileStatus) {
+      ((OzoneFileStatus) status)
+          .makeQualified(uri, path,
+              username, username);
+    }
+  }
+
+  @Override
+  public Iterator<BasicKeyInfo> listKeys(String pathStr) throws IOException {
+    incrementCounter(Statistic.OBJECTS_LIST);
+    OFSPath ofsPath = new OFSPath(pathStr);
+    String key = ofsPath.getKeyName();
+    OzoneBucket bucket = getBucket(ofsPath, false);
+    return new IteratorAdapter(bucket.listKeys(key));
+  }
+
+  public List<FileStatusAdapter> listStatus(String pathStr, boolean recursive,
+      String startPath, long numEntries, URI uri,
+      Path workingDir, String username) throws IOException {
+
+    OFSPath ofsPath = new OFSPath(pathStr);
+    String keyName = ofsPath.getKeyName();
+    OFSPath ofsStartPath = new OFSPath(startPath);
+    String startKey = ofsStartPath.getKeyName();
```

Comment:
Since I made all the input path params (`pathStr`) for `BasicRootedOzoneClientAdapterImpl` methods full path (volume+bucket+key), I believe it's best to keep `startPath` consistent. - That's why I intentionally renamed the parameter name in API definition from `startKey` to `startPath`.
 
Totally agree that doc is required. This should've been done in o3fs `BasicOzoneClientAdapterImpl` as well.

Reply:
Done in b21005ebacf25f5a3a6bfdb97348a2e2e567423d.